src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -85,40 +85,60 @@ static PSVersionInfo()
             //
             // The strings can be one of the following format examples:
             //    when powershell is built from a commit:
-            //      productVersion = '6.0.0-beta.7 Commits: 29 SHA: 52c6b...' convert to GitCommitId = 'v6.0.0-beta.7-29-g52c6b...'
-            //                                                                           PSVersion   = '6.0.0-beta.7'
+            //      productVersion = '7.3.0-preview.7 Commits: 29 SHA: 52c6b...' convert to GitCommitId = 'v7.3.0-preview.7-29-g52c6b...'
+            //                                                                           PSVersion   = '7.3.0-preview.7'
             //    when powershell is built from a release tag:
-            //      productVersion = '6.0.0-beta.7 SHA: f1ec9...'             convert to GitCommitId = 'v6.0.0-beta.7'
-            //                                                                           PSVersion   = '6.0.0-beta.7'
+            //      productVersion = '7.3.0-preview.7 SHA: f1ec9...'             convert to GitCommitId = 'v7.3.0-preview.7'
+            //                                                                           PSVersion   = '7.3.0-preview.7'
             //    when powershell is built from a release tag for RTM:
-            //      productVersion = '6.0.0 SHA: f1ec9...'                    convert to GitCommitId = 'v6.0.0'
-            //                                                                           PSVersion   = '6.0.0'
-            string rawGitCommitId;
-            string mainVersion = ProductVersion.Substring(0, ProductVersion.IndexOf(' '));
+            //      productVersion = '7.3.0 SHA: f1ec9...'                    convert to GitCommitId = 'v7.3.0'
+            //                                                                           PSVersion   = '7.3.0'
+            int index = ProductVersion.IndexOf(' ');
+            ReadOnlySpan<char> mainVersion = ProductVersion.AsSpan(0, index);
 
-            if (ProductVersion.Contains(" Commits: "))
-            {
-                rawGitCommitId = ProductVersion.Replace(" Commits: ", "-").Replace(" SHA: ", "-g");
-            }
-            else
-            {
-                rawGitCommitId = mainVersion;
-            }
+            string rawGitCommitId = ProductVersion.AsSpan(index).StartsWith(" Commits: ")
+                ? "v" + ProductVersion.Replace(" Commits: ", "-").Replace(" SHA: ", "-g")
```

Comment:
No optimization for private builds, intentionally.

Reply:
We have an issue that semicolon (`SHA:`) breaks WinForm. 
We got the format from early .Net verstions. Now they use simple hex git commit number format. I suggest to switch to the format too.

----

We never used `v` prefix from 6.0 version to current.

Reply:
> We have an issue that semicolon (SHA:) breaks WinForm.

Can you please point to the issue? At any rate, it's outside the scope of this PR. We can discuss whether any work should be done in the issue.

> We never used `v` prefix from 6.0 version to current.

Yeah, I was surprised that I didn't notice this :) Not sure how many people noticed it.
It should have the `v` prefix because our tag has it, which is the commit id for a specific release.
But I'm fine if the committee decides to reject this change.

Reply:
Reverted the `v` prefix change from this PR. Will submit a separate one so that it doesn't complicate things in this PR.

Reply:
> Can you please point to the issue?

#18131

My proposal is to continue to follow .Net and use something like
`7.0.0-preview.6.22351.2+f6720164c48ea0de634e574de94dd8450f761b25`
instead of 
`7.0.0-preview.6.22351.2 SHA: f6720164c48ea0de634e574de94dd8450f761b25`

> Yeah, I was surprised that I didn't notice this :) The example comments right above actually uses `v` prefix for commit id. Not sure how many people noticed it.
> It should have the `v` prefix because our tag has it, which is the commit id for a specific release.
> But I'm fine if the committee decides to reject this change.

The comment is outdated 😕 . Our conclusion at 6.0 pwsh time was to follow .Net pattern and they did not have the prefix. The prefix was removed from the code and from _csproj/prop_ files too. Comment in PowerShell.prop says that Windows Update (can) depend on this.
Today I suggest to follow new .Net pattern (used from .Net 3.0) and use simple hex commit id.

Reply:
> Our conclusion at 6.0 pwsh time was to follow .Net pattern and they did not have the prefix. The prefix was removed from the code and from csproj/prop files too.

I was not talking about the `ProductVersion`, which is fine to not have the prefix `v`. I was talking about our `GitCommitId` member in `$PSVersionTable`. We control that member and it's supposed to be a commit id. Without `v`, it's not a commit id.

Reply:
> Without `v`, it's not a commit id.

Commit id is unique SHA-1 hash. It can not have `v` prefix. Or do we say about different thing?

Reply:
For preview release or stable release, the commit id should be the release tag, which is `v7.3.0-preview.x` or `v7.3.0`. But `$PSVersionTable.GitCommitId` shows `7.3.0-preivew.x` or `7.3.0`, which is not the release tag.